### ISSUE TITLE
[Request]Can CPS changer be supported?

### ISSUE THREAD

**Issue by SumavisionQ5** [libretro/mame2003-plus-libretro#378]
There are 2 CPS changer games (Street Fighter Zero, Tenchi wo Kurau II/Warriors of Fate) dumped already and they are supported by FBA and latest MAME. 
SFZ is a downgraged port for CPS1, but for Tenchi wo Kurau II, it is a enhanced verion by CAPCOM and later ported to PlayStation and SEGA Saturn, which is only published in Japan. It is very popular in Asia and the game itself adds many new moves and a new mode. It is great worthwhile to play its secret mode.
Can these 2 games be support by MAME2003 plus?

**Comment by grant2258** [comment]
there two games are alread in mame2003 plus

**Comment by arcadez** [comment]
I've backported support for a few CPS2 games that were not playable in this core, they all had
one thing in common however in that they were added to MAME before the major changes to how
the protection decryption was handled.

In this ole core we are still using the XOR dumps while in latest MAME and FBA they are using
keys for example.........

MAME78 19xx

	ROM_REGION16_BE( CODE_SIZE, REGION_USER1, 0 )
	ROM_LOAD16_WORD_SWAP( "19xux.03", 0x000000, 0x80000, CRC(239a08ae) SHA1(a57e5a0cef351485569cf5c3c79ca0cc7ce439cf) )
	ROM_LOAD16_WORD_SWAP( "19xux.04", 0x080000, 0x80000, CRC(c13a1072) SHA1(e5c1c85cb77d69d5f43f8cfa6bcc894bd31e0b68) )
	ROM_LOAD16_WORD_SWAP( "19xux.05", 0x100000, 0x80000, CRC(8c066ec3) SHA1(a540a773e1cb04a814feb4be8cf22b0eb3121cee) )
	ROM_LOAD16_WORD_SWAP( "19xux.06", 0x180000, 0x80000, CRC(4b1caeb9) SHA1(8f965054e5b1c3f1dcaa20279a3348d63a94c1d6) )


Latest MAME same rom 19xx

ROM_REGION( 0x20, "key", 0 )
ROM_LOAD( "19xx.key", 0x000000, 0x000014, CRC(77e67ba1) SHA1(8c995bd00941ffe0145cb80e4f105ef8ef0a0001) )

In laymans terms what this means is the whole Capcom core in MAME78 would have to be updated
to support the newer decryption style and all the CPS2 roms would have to change in the process i seriously doubt there would be much support for this for the addition of a few extra versions of Street Fighter

However if the versions you want to add were unprotected and needed no decryption then it
might be possible.

**Comment by arcadez** [comment]
With some further investigation it seems the CPS Changer version of Street Fighter Zero was
supported a long time ago then it got removed.........

- 0.36RC1: This game was removed from MAME. Nicola wrote: One of the removed drivers is Street Fighter Zero for the CPS Changer, which is not an arcade game even if it runs on arcade hardware. The source is left in, commented out, because it could be useful as a reference for people attempting to emulate CPS-2 (which, before you ask, won't happen anytime soon).

- 0.35fix: Fixed dipswitches in Street Fighter ZERO.

- 0.35: Added Street Fighter ZERO (Japan CPS Changer) (Capcom 1995).

**Comment by SumavisionQ5** [comment]
@arcadez @grant2258 
The CPS changer version of Tenchi wo Kurau is quite different from CPS version. It has a secret mode and the way to enable secret mode will be told if you finish the game in 1 coin. And each character has a new combo and move. Its romname is wofch.zip, and is not supported by MAME2003 plus now.
About Street Fighter ZERO (Japan CPS Changer), it is a cheap port for home console, and not as good as CPS2 version.

**Comment by arcadez** [comment]
Alright i have it up and running in my test core everything seems fine i'll port the code
across later

**Comment by arcadez** [comment]
Request fulfilled

**Comment by markwkidd** [comment]
what a deal! thanks arcadez

**Comment by grant2258** [comment]
nice one again mate i never even knew this system existed. Speaking of cps2 we could add the decrypted roms to the ones that arent working, its just the issue that people would need to find the roms

**Comment by Wilstorm** [comment]
@arcadez - Thanks for the new games!

**Comment by arcadez** [comment]
@grant2258 it might be worth fixing a game if the parent rom is non working there are only
two in MAME78........

Marvel Super Heroes (Euro 951024)
Vampire Hunter 2 - Darkstalkers Revenge (Japan 970929)

**Comment by arcadez** [comment]
Well whatya know we cant actually do the above as the decryption dumps are not compatable
with our ole CPS2 code that's that then i guess :)

**Comment by grant2258** [comment]
i should have guessed it was just an idea soz for the time waste. I didnt think there would be an issue 
using decrypted roms
